web-local/src/lib/svelte-query/invalidation.ts

```diff
@@ -90,3 +94,28 @@ export function invalidateProfilingQueries(
     },
   });
 }
+
+export const removeModelQueries = async (
+  queryClient: QueryClient,
+  instanceId: string,
+  name: string
+) => {
+  const path = getFilePathFromNameAndType(name, EntityType.Model);
+
+  // remove affected catalog entries and files
+  await Promise.all([
+    queryClient.removeQueries(
+      getRuntimeServiceGetFileQueryKey(instanceId, path)
+    ),
+    queryClient.removeQueries(
+      getRuntimeServiceGetCatalogEntryQueryKey(instanceId, name)
+    ),
+  ]);
+
+  // remove profiling queries
+  return queryClient.removeQueries({
```

Comment:
We should run this for metrics as well. Do something similar to `getInvalidationsForPath` from `invalidations.ts`